pkg/webhook/util/filter.go

```diff
@@ -0,0 +1,25 @@
+package util
+
+import (
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	ctlbackup "github.com/harvester/harvester/pkg/controller/master/backup"
+	ctlharvesterv1 "github.com/harvester/harvester/pkg/generated/controllers/harvesterhci.io/v1beta1"
+	"github.com/harvester/harvester/pkg/webhook/indexeres"
+)
+
+func HasInProgressingVMBackupBySourceUID(cache ctlharvesterv1.VirtualMachineBackupCache, sourceUID string) (bool, error) {
+	vmBackups, err := cache.GetByIndex(indexeres.VMBackupBySourceUIDIndex, sourceUID)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	for _, vmBackup := range vmBackups {
```

Comment:
One reliability question:
If one VM`s backup takes long time and not finished (due to various of possible reasons), but the user is urgent to restart the VM, is it possible to delete/stop the related vm backup, then restart the VM?

The assumption is: VM stop/restart is in higher priority than VM backup, the backup needs to give way for user to end backup asap, 

at least, avoid the potential dead loop: backup is never ending (due to bug ...), stop/restart VM is also not allowed.

Reply:
@w13915984028, my thought is to let users remove error/progressing backup before stopping/restarting a VM. If VM stop/restart is in higher priority, we may need a setting for users to choose whether ignore the webhook.
@guangbochen WDYT? Thank you.

Reply:
I think it makes sense to ask users to remove the backup first if VM restart or shutdown action is required during the backup. Also, we need to make sure not to block deleteing the backup.